llvm/lib/Transforms/Utils/SimplifyCFG.cpp

```diff
@@ -4074,6 +4084,13 @@ static bool performBranchToCommonDestFolding(BranchInst *BI, BranchInst *PBI,
   Value *BICond = VMap[BI->getCondition()];
   PBI->setCondition(
       createLogicalOp(Builder, Opc, PBI->getCondition(), BICond, "or.cond"));
+  if (!ProfcheckDisableMetadataFixes)
+    if (auto *SI = dyn_cast<SelectInst>(PBI->getCondition()))
+      if (!MDWeights.empty()) {
+        assert(isSelectInRoleOfConjunctionOrDisjunction(SI));
```

Comment:
Note: I preferred `assert`-ing here (and below) because the expectation is the condition comes from `createLogicalOp` and the only reason we get a `select` is because of how `or` or `and` get lowered there.